packages/mobx/src/types/observableset.ts

```diff
@@ -243,6 +243,58 @@ export class ObservableSet<T = any> implements Set<T>, IInterceptable<ISetWillCh
         } as any)
     }
 
+    intersection<U>(otherSet: ReadonlySetLike<U> | Set<U>): Set<T & U> {
+        if (isES6Set(otherSet)) {
+            this.atom_.reportObserved()
+            return otherSet.intersection(this.data_)
```

Comment:
This isn't correct, it won't work with custom enhancer. Was there any problem with the code I've proposed originally? That is:
```ts
if (isES6Set(otherSet)) {
   return otherSet.intersection(this)
} else {
  // ...
}
```

Reply:
I've checked twice where you mentioned about deleting `reportObserved()` but didn't find. Maybe you're assumed this but I didn't get it. So anyway removing the `reportObserved()` throwing errors:

![image](https://github.com/mobxjs/mobx/assets/1235286/9374cec5-da77-450b-8e6b-64485806415c)

Reply:
> where you mentioned about deleting reportObserved() but didn't find.

I didn't mention deleting `reportObserved` anywhere, because I didn't propose adding it in the first place:
https://github.com/mobxjs/mobx/pull/3853#issuecomment-2033993906

> So anyway removing the reportObserved() throwing errors:

Did you just removed `reportObserved` or did you also replaced `_data` with `this` as suggested?

Reply:
Oh, I've tought you're mentioned it in your latest messages. Totally forgot about this one! In this case everything is okay